src/rf/apps/core/models.py

```diff
@@ -133,6 +135,30 @@ class Meta:
     thumb_large_key = CharField(max_length=255, blank=True, default='',
                                 help_text='S3 key for large thumbnail')
 
+    def retry_possible(self):
+        """
+        Returns true if it is possible to retry processing a layer.
+        """
+        return self.status == enums.STATUS_FAILED
```

Comment:
Ok. I just imagine someone uploading junk files and hitting the retry button a hundred times.